### PR TITLE
Add vertual environment in the linux runner for running tests

### PR DESCRIPTION
**Pull request by swapneelap**

### CODE REVIEW

**Comment by lang-m** [comment]
@swapneelap We should consider using [tox](https://tox.wiki/en/latest/) as an alternative before you put too much effort into this. I think tox could significantly simplify our problem.

**Comment by swapneelap** [comment]
@lang-m Have a look at the workflow file now. All the commands are executed in `test-env` conda environment and the step for removing the `test-env` runs regardless of the above tests pass or not.

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -73,34 +73,49 @@ jobs:
...
-    - name: Install package
-      run: python -m pip install .[dev]
-
-    - name: Unit tests
-      run: python -m pytest -v mumax3c
+    - name: Test mumax3c
+      run: |
+        eval "$(conda shell.bash hook)"
+        conda activate test-env
+        python -m pytest -v mumax3c
```
Comment (lang-m) [inline .github/workflows/workflow.yml:96]: We should switch back to using `invoke` here and I would suggest to test coverage on the Linux runner.

Comment (swapneelap) [reply .github/workflows/workflow.yml]: I have made the changes, have a look. Do you think we should add `if: always()` to test coverage even if other tests fail?

Comment (swapneelap) [reply .github/workflows/workflow.yml]: @lang-m I understand. I have made the required changes, have a look.

Comment (lang-m) [reply .github/workflows/workflow.yml]: Looks good

**Review by lang-m — APPROVED** [review]

**Pull request merged**